lib/inv_gcr_quda.cpp

```diff
@@ -323,34 +345,39 @@ namespace quda {
 
       // update since n_krylov or maxiter reached, converged or reliable update required
       // note that the heavy quark residual will by definition only be checked every n_krylov steps
-      if (k == n_krylov || total_iter == param.maxiter || (r2 < stop && !l2_converge) || sqrt(r2 / r2_old) < param.delta) {
+      if (k == n_krylov || total_iter == param.maxiter || (r2[0] < stop[0] && !l2_converge)
+          || sqrt(r2[0] / r2_old[0]) < param.delta) {
 
         // update the solution vector
-        updateSolution(x, alpha, beta, gamma, k, p);
+        for (auto i = 0u; i < b.size(); i++) updateSolution(x[i], alpha[i], beta[i], gamma[i], k, get_i(p, i));
```

Comment:
Why loop instead of an MRHS udpateSolution here?

Reply:
This is because the multi-blas routines are not yet batch aware (which are used in `updateSolution`).  This is left as a future optimization for some PR in the future.